api/cmd/main.go

```diff
@@ -0,0 +1,177 @@
+package main
+
+import (
+	"fmt"
+	"math/big"
+	"net/http"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	"github.com/project-nova/backend/pkg/database"
+	"github.com/thirdweb-dev/go-sdk/v2/thirdweb"
+)
+
+type WhitelistWallet struct {
+	ID          string `gorm:"primaryKey;column:id"`
+	Address     *string
+	MerkleProof *string
+	CreatedAt   time.Time
+}
+
+type StoryNFT struct {
+	Name   string
+	Symbol string
+	Count  uint64
+	URI    string
+}
+
+func (WhitelistWallet) TableName() string {
+	return "whitelist_wallet"
+}
+
+func main() {
+	r := gin.Default()
+
+	r.GET("/mint/proof", func(c *gin.Context) {
+		address := c.DefaultQuery("address", "")
+
+		// Validate address
+		if address == "" {
+			c.String(http.StatusBadRequest, fmt.Sprintf("input address is invalid, address: %s", address))
+			return
+		}
+
+		db, err := database.NewGormDB("postgresql://postgres:@localhost:23005/postgres?sslmode=disable")
+		if err != nil {
+			c.String(http.StatusInternalServerError, "Internal server error")
+			return
+		}
+
+		result := &WhitelistWallet{}
+		r := db.Where("address = ?", address).First(&result)
+		if r.Error != nil {
+			c.String(http.StatusBadRequest, fmt.Sprintf("address is invalid: %v", r.Error))
+			return
+		}
+
+		c.JSON(http.StatusOK, gin.H{
+			"proof": *result.MerkleProof,
+		})
+	})
+
+	r.GET("/nfts", func(c *gin.Context) {
+		privateKey := "a6cd3f393b1cddf8be66e2ff784640adbafbce852267ec1ec0e22eb741232000"
+
+		sdk, err := thirdweb.NewThirdwebSDK("goerli", &thirdweb.SDKOptions{
+			PrivateKey: privateKey,
+		})
+		if err != nil {
+			fmt.Printf("Failed to set up sdk: %v", err)
+			c.String(http.StatusInternalServerError, "Internal server error")
+			return
+		}
+
+		// You can replace your own contract address here
+		contractAddress := "0xbcF76d7B52D6edef3D6Ec009D0371F06895B4E6A"
```

Comment:
The latest contract address is deployed at: 0x64432E5A76a93e79be2f7F3F12982059a32Fd794

Reply:
Updated. Thanks